PUML.h

```diff
@@ -780,17 +820,11 @@ class PUML
 		m_faces.clear();
 		m_v2e.clear();
 
-		unsigned long cellOffset = m_originalSize[0];
-#ifdef USE_MPI
-		MPI_Scan(MPI_IN_PLACE, &cellOffset, 1, MPI_UNSIGNED_LONG, MPI_SUM, m_comm);
-#endif // USE_MPI
-		cellOffset -= m_originalSize[0];
-
 		std::vector<std::set<unsigned int> > edgeUpward;
 		std::set<unsigned int>* vertexUpward = new std::set<unsigned int>[m_vertices.size()];
 
 		for (unsigned int i = 0; i < m_originalSize[0]; i++) {
-			m_cells[i].m_gid = i + cellOffset;
+			m_cells[i].m_gid = cellDistributor.localToGlobalId(rank, i);
```

Comment:
Could this change actually fix some MPI errors we had before?

Reply:
Most errors should have already been fixed with https://github.com/TUM-I5/PUML2/pull/6, but yes, this could fix some more.